server/src/entities/test_result.ts

```diff
@@ -8,3 +9,5 @@ export type TestResult = {
     testDate: Date;
     addressId: number;
 };
+
+export type TestResultWithAddress = TestResult & Address;
```

Comment:
I would probably just update the test result entity to include the address as an attribute same way we do with the user or at the very least make it so this has address as an attribute instead of a union so the attributes aren't all on the same level.

Also just realized the `TestResult` doesnt contain the testId right now.

```suggestion
export type TestResult = {
    testId: number
    patientId: number;
    result: TestResultType;
    testType: TestType;
    testDate: Date;
    address: Address;
};
```

Reply:
the reason why im creating a new type is because TestResult is used for the post which wouldn't have the full address or the testId so in the case of only having one type I would have to leave some attributes null. 

Would that still be a valid use of the type?

Reply:
Yea its fine to leave attributes null in different scenarios like the POST.
I think in general its best to be consistent with the types so a TestResult is always a TestResult and then its up to the function to determine if the values are null or not.

This is generally how almost all ORMs and large enterprise style web frameworks like Java Spring and .NET for C# handle types

Reply:
ayt will change it then!